nnsvs/bin/train.py

```diff
@@ -160,25 +161,29 @@ def train_loop(config, device, model, optimizer, lr_scheduler, data_loaders):
                 optimizer.zero_grad()
 
                 # Run forwaard
-                y_hat = model(x, sorted_lengths)
-
-                # Compute loss
-                mask = make_non_pad_mask(sorted_lengths).unsqueeze(-1).to(device)
-
-                if config.train.stream_wise_loss:
-                    # Strean-wise loss
-                    streams = split_streams(y, config.model.stream_sizes)
-                    streams_hat = split_streams(y_hat, config.model.stream_sizes)
-                    loss = 0
-                    for s_hat, s, sw in zip(streams_hat, streams, stream_weights):
-                        s_hat_mask = s_hat.masked_select(mask)
-                        s_mask = s.masked_select(mask)
-                        loss += sw * criterion(s_hat_mask, s_mask).mean()
+                if model.prediction_type == "probabilistic":
+                    pi, sigma, mu = model(x, sorted_lengths)
+                    loss = mdn_loss(pi, sigma, mu, y).mean()
```

Comment:
One more thing I just noticed here: could you add mask here? If the batch size is larger than 1, we'd need to care about padding part. I'd guess it doesn't significantly change the performance, but it shouldn't harm and is more appropriate.

Reply:
I understand it, but please give me some time for me to come up with a good way.  I wrote mdn.mdn_loss as it makes use of the same data structure which MDNLayer does, and I don't have any good idea to handle the return value of masked_select(1d Tensor).

Reply:
MDN model does not output predictions of y directly, so I mask the output of mdn_loss according to the shape of  y like this;

                if model.prediction_type == "probabilistic":
                    pi, sigma, mu = model(x, sorted_lengths)

                    # (B, max(T))
                    mask = make_non_pad_mask(sorted_lengths).to(device)
                    # Compute loss and apply mask
                    loss = mdn_loss(pi, sigma, mu, y, reduce=False).masked_select(mask).mean()

Please advise me if there is better way to do.